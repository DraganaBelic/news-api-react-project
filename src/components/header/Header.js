import React, { useState } from "react";
import PropTypes from "prop-types";
import { HeaderStyle, HeaderItemStyle } from "./HeaderStyle";
import { setCountry } from "../../actions/countryActions";
import { country } from "../../constants/Constant";
import { useDispatch, useSelector } from "react-redux";
import { NavLink, withRouter } from "react-router-dom";
import { useTranslation } from "react-i18next";

const Header = (props) => {
  const dispatch = useDispatch();
  const [visited, setVisited] = useState(country.GB);
  const countryState = useSelector((state) => state.country);
  const { t } = useTranslation();

  const currentCountry =
    countryState && countryState.currentCountry
      ? countryState.currentCountry
      : country.GB;
  const disabled = useSelector((state) => state.country.disable);
  const handleClick = (chosenCountry) => {
    if (!disabled) {
      dispatch(setCountry(chosenCountry));
      setVisited(chosenCountry);
    }
  };
  const { location } = props;
  const changeCountryInPath = (chosenCountry) => {
    let newPath = "";
    if (chosenCountry !== currentCountry) {
      newPath = location.pathname.replace(currentCountry, chosenCountry);
    }
    return newPath;
  };
  return (
    <HeaderStyle>
      <NavLink exact to={`/${currentCountry}`}>
        <HeaderItemStyle>{t("topNews")}</HeaderItemStyle>
      </NavLink>
      <NavLink to={`/${currentCountry}/categories`}>
        <HeaderItemStyle>{t("categories")}</HeaderItemStyle>
      </NavLink>
      <NavLink to={`/${currentCountry}/search`}>
        <HeaderItemStyle>{t("search")}</HeaderItemStyle>
      </NavLink>
      <NavLink
        to={() => changeCountryInPath(country.US)}
        onClick={() => handleClick(country.US)}
      >
        <HeaderItemStyle
          disabled={disabled}
          active={visited === country.US}
          small
          left_border
          right
        >
          <div>US</div>
        </HeaderItemStyle>
      </NavLink>
      <NavLink
        to={() => changeCountryInPath(country.GB)}
        onClick={() => handleClick(country.GB)}
      >
        <HeaderItemStyle
          disabled={disabled}
          active={visited === country.GB}
          small
          right
        >
          <div>GB</div>
        </HeaderItemStyle>
      </NavLink>
    </HeaderStyle>
  );
};

Header.propTypes = {
  disabled: PropTypes.bool,
};

export default withRouter(Header);
