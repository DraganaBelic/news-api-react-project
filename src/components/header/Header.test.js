import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import thunk from "redux-thunk";
import { MemoryRouter } from "react-router-dom";
import Header from "./Header";

const middlewares = [thunk];
const initialState = {
  country: {
    countryState: "gb",
  },
};
export const mockedStore = (state = initialState) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};
jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (s) => s }),
}));
describe("Test Header component", () => {
  it("should render properly", () => {
    const store = mockedStore();
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
