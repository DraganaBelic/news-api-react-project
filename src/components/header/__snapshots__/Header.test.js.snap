// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component should render properly 1`] = `
<div
  class="HeaderStyle-xeym88-0 TXrIz"
>
  <a
    href="/gb"
  >
    <div
      class="HeaderStyle__HeaderItemStyle-xeym88-1 dvtrWk"
    >
      topNews
    </div>
  </a>
  <a
    href="/gb/categories"
  >
    <div
      class="HeaderStyle__HeaderItemStyle-xeym88-1 dvtrWk"
    >
      categories
    </div>
  </a>
  <a
    href="/gb/search"
  >
    <div
      class="HeaderStyle__HeaderItemStyle-xeym88-1 dvtrWk"
    >
      search
    </div>
  </a>
  <a
    aria-current="page"
    class="active"
    href="/"
  >
    <div
      class="HeaderStyle__HeaderItemStyle-xeym88-1 kwIewx"
    >
      <div>
        US
      </div>
    </div>
  </a>
  <a
    aria-current="page"
    class="active"
    href="/"
  >
    <div
      class="HeaderStyle__HeaderItemStyle-xeym88-1 eCcOIj"
    >
      <div>
        GB
      </div>
    </div>
  </a>
</div>
`;
